### ISSUE TITLE
knex migrate:latest --fake

### ISSUE THREAD

**Issue by schuyler1d** [tgriesser/knex#3303]
# Environment

Knex version:0.17.2
Database + version: generic, but postgres, sqlite
OS: linux

# Feature discussion / request

1. Explain what is your use case
Instead of starting from scratch, it would be good to be able to set migrations to a certain point without running the migrations themselves -- that way we can get an existing DB that has already updated state, but not with knex migrate, setup to use knex migrate in the future.

2. Explain what kind of feature would support this
Django ORM has this feature where you can run with a `--fake` option and it updates the equivalent of the knex_migrations table with whatever should have happened.

3. Give some API proposal, how the feature should work
e.g. `knex migrate:latest --fake` or `knex migrate:down --fake` would update the knex_migrations table without running the `up` or `down` methods.

**Comment by kibertoad** [comment]
Sounds useful. PR would be most welcome!

**Comment by cosileone** [comment]
Would also like to add that `--fake` flag would be useful on specific migrations as well so:
```
knex migrate:make migration_name --fake
```

**Comment by LeonardoDiehl** [comment]
I would also like to see this implemented! If you're willing to do it by yourself please feel free to call me for help.

**Comment by mercmobily** [comment]
Agreed - a fake mode is very helpful for adopting knex on an existing DB.

We could add a `--fake` flag to `migrate:latest` and `migrate:down` that records the migration(s) in `knex_migrations` without executing their `up`/`down` bodies. This mirrors Django’s approach and is safe when paired with careful use.

Implementation sketch:
- CLI: parse `--fake` and pass it through to the migrator.
- Migrator: in lib/migrations/migrate/Migrator.js, branch in the execution path to only insert/delete rows in the migrations tables when `fake` is true, and log the names accordingly.
- Guardrails: refuse `--fake` if a migration has already run, and print a clear banner summarising what will be marked.
